nomos-core/chain-defs/src/mantle/genesis_tx.rs

```diff
@@ -0,0 +1,391 @@
+            inscription: vec![1, 2, 3, 4],
+            parent,
+            signer,
+        }
+    }
+
+    fn sdp_declare_op(
+        utxo_to_use: Utxo,
+        zk_id_value: u8,
+        verifying_key: VerifyingKey,
+    ) -> SDPDeclareOp {
+        SDPDeclareOp {
+            service_type: ServiceType::BlendNetwork,
+            locked_note_id: utxo_to_use.id(),
+            zk_id: ZkPublicKey(BigUint::from(zk_id_value).into()),
+            provider_id: ProviderId(verifying_key),
+            locators: [].into(),
+        }
+    }
+
+    fn blob_op(channel_id: ChannelId, verifying_key: VerifyingKey) -> BlobOp {
+        BlobOp {
+            channel: channel_id,
+            blob: [42; 32],
+            blob_size: 1024,
+            da_storage_gas_price: 10,
+            parent: MsgId::root(),
+            signer: verifying_key,
+        }
+    }
+
+    // Helper function to create a test note
+    fn create_test_note(value: Value) -> Note {
+        Note::new(value, PublicKey::from(BigUint::from(123u64)))
+    }
+
+    // Helper function to create a basic signed transaction
+    fn create_signed_tx(ops: Vec<Op>) -> SignedMantleTx {
+        let ledger_tx = LedgerTx::new(vec![], vec![create_test_note(1000)]);
+        let ops_proofs_len = ops.len();
+        let mantle_tx = MantleTx {
+            ops,
+            ledger_tx,
+            execution_gas_price: 0,
+            storage_gas_price: 0,
+        };
+
+        SignedMantleTx {
+            mantle_tx,
+            ops_proofs: vec![None; ops_proofs_len],
+            ledger_tx_proof: DummyZkSignature::prove(ZkSignaturePublic {
+                msg_hash: Fr::from(0u64),
+                pks: vec![],
+            }),
+        }
+    }
+
+    #[test]
+    fn test_inscription_fields() {
+        // check inscription with channel id [1; 32] fails
+        let tx = create_signed_tx(vec![Op::ChannelInscribe(inscription_op(
+            ChannelId::from([1; 32]),
+            MsgId::root(),
+            VerifyingKey::from_bytes(&[0; 32]).unwrap(),
+        ))]);
+        assert!(matches!(
+            GenesisTx::from_tx(tx),
+            Err(Error::InvalidInscription(_))
+        ));
+
+        // check inscription with non-root parent fails
+        let tx = create_signed_tx(vec![Op::ChannelInscribe(inscription_op(
+            ChannelId::from([0; 32]),
+            MsgId::from([1; 32]),
+            VerifyingKey::from_bytes(&[0; 32]).unwrap(),
+        ))]);
+        assert!(matches!(
+            GenesisTx::from_tx(tx),
+            Err(Error::InvalidInscription(_))
+        ));
+
+        // TODO: check inscription with non-zero signer fails
+        let tx = create_signed_tx(vec![Op::ChannelInscribe(inscription_op(
+            ChannelId::from([0; 32]),
+            MsgId::root(),
+            VerifyingKey::from_bytes(&[1; 32]).unwrap(),
+        ))]);
+        assert!(matches!(
+            GenesisTx::from_tx(tx),
+            Err(Error::InvalidInscription(_))
+        ));
```

Comment:
is the `TODO` done? it looks like this test is checking the non-zero signer case